pyanaconda/payload/livepayload.py

```diff
@@ -166,14 +166,25 @@ def install(self):
         threadMgr.wait(THREAD_LIVE_PROGRESS)
 
         # Live needs to create the rescue image before bootloader is written
-        if not os.path.exists(util.getSysroot() + "/usr/sbin/new-kernel-pkg"):
-            log.error("new-kernel-pkg does not exist - grubby wasn't installed?  skipping")
-            return
+        if os.path.exists(util.getSysroot() + "/usr/sbin/new-kernel-pkg"):
+            useNKP = True
+        else:
+            log.warning("new-kernel-pkg does not exist - grubby wasn't installed?")
+            useNKP = False
 
         for kernel in self.kernelVersionList:
             log.info("Generating rescue image for %s", kernel)
-            util.execInSysroot("new-kernel-pkg",
-                               ["--rpmposttrans", kernel])
+            if useNKP:
+                util.execInSysroot("new-kernel-pkg",
+                                   ["--rpmposttrans", kernel])
+            else:
+                files = glob.glob(util.getSysroot() + "/etc/kernel/postinst.d/*")
+                srlen = len(util.getSysroot())
+                files = sorted([f[srlen:] for f in files
+                                if os.access(f, os.X_OK)])
+                for file in files:
+                    util.execInSysroot(file,
+                                       [kernel, "/boot/vmlinuz-%s" % kernel])
```

Comment:
The standard way to install kernels is to invoke `/bin/kernel-install`. `kernel-install` was written by dracut author to provide a uniform interface for kernel installation and so that random tools don't try to implement all the logic. Finding and calling the `postinst.d` scripts like that looks very iffy. @haraldh comments?

Reply:
Yes, the kernel-core package also calls ```kernel-install```:
```
preuninstall scriptlet (using /bin/sh):
/bin/kernel-install remove 4.18.14-200.fc28.x86_64 /lib/modules/4.18.14-200.fc28.x86_64/vmlinuz || exit $?
posttrans scriptlet (using /bin/sh):
/bin/kernel-install add 4.18.14-200.fc28.x86_64 /lib/modules/4.18.14-200.fc28.x86_64/vmlinuz || exit $?
```
Why call the scripts manually?

Reply:
@keszybz @haraldh I agree for kernel installs, but in this case the code doesn't do a kernel install, it just executes `new-kernel-pkg --rpmposttrans`. In the `new-kernel-pkg` script that is:

```
rpmposttrans()
{
    local files
    local f
    files="/etc/kernel/postinst.d/*[^~] /etc/kernel/postinst.d/$version/*[^~]"
    for f in $files ; do
        [ -f $f ] && [ -x $f ] || continue
        $f $version $kernelImage
    done
}
```

And that's why the `/etc/kernel/postinst.d/` scripts are called directly when `new-kernel-pkg` doesn't exists.

I'm not that familiar with this part of the Anaconda code to comment if a whole `kernel-install` is suitable here instead. But that will certainly be a change in behaviour with respect to just calling `kernel-pkg --rpmposttrans` as it's the case when `new-kernel-pkg` is available.

I'll let @vathpela comment on this since he is more familiar with Anaconda and this is part of his patch.